src/onnx-mlir.cpp

```diff
@@ -68,7 +69,23 @@ int main(int argc, char *argv[]) {
   }
 
   // Create context after MLIRContextCLOptions are registered and parsed.
+  // The multi-threading in MLIRContext is enabled by default. It must be
+  // disabled to control the number of threads. To use single thread, simply
+  // disable it. To use a specific number of threads, disable it once and then
+  // set a new thread pool.
   mlir::MLIRContext context;
+  std::unique_ptr<llvm::ThreadPoolInterface> threadPoolPtr;
+  assert(compilationNumThreads >= 0 &&
+         "The number of threads for compilation by "
+         "'-j' option should be positive value.");
```

Comment:
You may define `compilationNumThreads` to be `uint64_t`, so that it is never negative, e.g. `%llvm::cl::opt<int, true> compilation_num_threads("j",`

Reply:
Done. Thanks!